org.gramps_project.Gramps.yml

```diff
@@ -0,0 +1,115 @@
+app-id: org.gramps_project.Gramps
+# flatpak-builder will not take com.gramps-project.Gramps
+runtime: org.gnome.Platform
+runtime-version: '3.38'
+sdk: org.gnome.Sdk
+command: /app/bin/gramps
+rename-icon: gramps
+finish-args:
+# needs home access to access documents and pictures
+  - --filesystem=home
```

Comment:
I'm not a reviewer but as a user, I would greatly appreciate a less permissive filesystem access.
```
  - --filesystem=xdg-documents
  - --filesystem=xdg-download
  - --filesystem=xdg-pictures
  - --persist=.gramps
```
Maybe you would be able to drop the persistent folder with wrapper script using setting `export GRAMPSHOME=$XDG_CONFIG_HOME/gramps`.

Reply:
Hi, I noticed that if restoring a backup in Gramps that contains media files, Gramps will make a new directory in the home directory for the media files, and the name varies depending on what the name of the backup is.  So it needs write access to the home directory for that.

Reply:
Please note: there is a template in the PR that asked this exact question. It would have been helpful to look through the questions there and proactively fill them out to reduce back and forth during review.

Reply:
@OzarkShepherd I just tested this and with `GRAMPSHOME` set the default location in actually under `GRAMPSHOME`.
![Swappshot Sun Dec 20 23:49:29 2020](https://user-images.githubusercontent.com/44030034/102725340-b46c2c00-431e-11eb-8c4a-59f21d54155c.png)

Reply:
@tinywrkb If I understand correctly, you want me to include another script and have the Gramps manifest call the init script during installation to change the default home location?  I don't know how to do this and I don't want to be unresponsive, but I am concerned about further complicating things in a way that will reduce functionality for some users, especially less advanced users.  For example, someone might make a new directory in home just for genealogy (which I have done in the past) and then get confused why the Gramps flatpak won't access it while the distro-installed or python-installed versions could access it.  Also some people with poor memory and concentration like me will see the altered base path and have trouble finding the media directory again once the close button is clicked.  I really do appreciate the help here and from reddit in the past, though.

Reply:
What about setting  `--env=GRAMPSHOME=~/Gramps --filesystem=~/Gramps:create`?
This way you have a dedicated folder for Gramps, in a visible location, not in a hidden folder, you don't give permissive access to `~`, and you don't even need an init script (and patching the desktop file).  
It still changes the default behavior but considering upstream didn't step in and don't follow the XDG dirs spec then it's not a bad workaround, just needs to be properly tested. I think it's a packaging decision left for the maintainer.    
You can add a comment to the application page on Flathub about the default Gramps folder location.  

I will add that this doesn't affect me, I have a global `--nofilesystem=home` override because I find it unacceptable permission.  
Anyway, this is all I have to say about this issue.

Reply:
@OzarkShepherd

The general idea is to keep application data as self-contained as possible, so `--persist=.gramps` is the bare minimum. By default, applications have write access to `$XDG_DATA_HOME` and it could be used as `$GRAMPSHOME` (`~/.var/app/org.gramps_project.Gramps/data`) instead; users can always modify these permissions with `flatpak override` or [Flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal).

I will make these changes for you if you don't mind.

Reply:
Ah well, I don't have write access to your repo.

You can run `curl https://paste.xinu.at/Xpbx4stc | git am` locally to apply my changes.

Reply:
> The general idea is to keep application data as self-contained as possible, so --persist=.gramps is the bare minimum. By default, applications have write access to $XDG_DATA_HOME and it could be used as $GRAMPSHOME (~/.var/app/org.gramps_project.Gramps/data) instead; users can always modify these permissions with flatpak override or Flatseal.

In a genealogy program like Gramps, users tend to download documents like pdf's and images of old documents like birth and death records to use as evidence.  Without this evidence to back up their claims, some people consider the genealogy claims to be spurious and useless at best, and misinformation at worst.  The program also organizes this evidence into a database by associating them with people and events so that they can be easily checked.  That is why the genealogy program Gramps must have access to the Documents, Downloads, and Pictures directories as a default, so that the evidence can be accessed.  I personally have my evidence in the Documents folder, others use Pictures or Downloads.  Gramps uses .gramps for the database, but not the pictures and other evidence.  The evidence is not duplicated into the database, only referenced; duplication only happens when the database is backed up with media files included.  My backups tend to be about 500 MB, and my wife's are much bigger, so we aren't just discussing a few pictures.  Furthermore, the Gramps flatpak has metadata capability so that notes can be made about the pictures, like what they are from and how they were obtained, so write permission is needed.

While advanced users can use flatseal or terminal commands to alter permissions, not everyone who uses flatpaks is an advanced user.  So that is not an option for many users.  I didn't even know about flatseal until I started to research how to make a flatpak.  They will just be stuck with an flatpak with reduced functionality from the older version that they could obtain from their distro.  The reason I got interested in making a flatpak for Gramps is for users like me who still want a fully functional and up to date genealogy app, but want to use a stable distribution so they don't have to deal with instability in their OS.  There is the option of installing dependencies and building Gramps from source, but that is too advanced for some users and since it is a python installation, removing the old python files when upgrading can get messy.  I still have Gramps 5.1.2 installed through python on my OpenSUSE system, even though the current version is 5.1.3.  And another popular stable distribution, Debian 10, through normal channels is Gramps 5.0.1 according to the gramps-project wiki, even though backports has 5.1.2 which is still outdated.

Reply:
All we can do is advise; as it's "yours" application, you get to decide what feels right.

Reply:
I'm not trying to be difficult, I just want Gramps to be functional and can't find any good explanation of xdg online through google, that will ensure that the users will have r/w access to the necessary Documents, Downloads, and Pictures.  That is why I resist making the change.  I have read the flatpak manual and saw that r/w is default for xdg but I don't understand the point.  I understand that flathub wants minimum permissions for the sandbox, but I suspect a lot of users might run into problems if they don't have access to their documents.  I did advise the gramps-project administrator about a week ago for the changes proposed by tinywrkb, but am waiting for a response.